lib/Service/MailTransmission.php

```diff
@@ -321,6 +337,29 @@ private function handleLocalAttachment(string $userId, array $attachment, IMessa
 		}
 	}
 
+	/**
+	 * @param string $userId
+	 * @param array $attachment
+	 * @param IMessage $message
+	 *
+	 * @return void
+	 *
+	 * Adds an attachment that's coming from another message's attachment (typical use case: email forwarding)
+	 *
+	 */
+	private function handleEmailAttachment(string $userId, array $attachment, IMessage $message) {
+
+		// Gets attachment from other message
+		$attachmentMessage = $this->mailManager->getMessage($userId, (int)$attachment['messageId']);
+		$mailbox = $this->mailManager->getMailbox($userId, $attachmentMessage->getMailboxId());
+		$account = $this->accountService->find($userId, $mailbox->getAccountId());
+		$folder = $account->getMailbox($mailbox->getName());
+		$attachment = $folder->getAttachment($attachmentMessage->getUid(), $attachment['id']);
```

Comment:
I'm having mixed feelings here

the "folder" code is the type of code I really want to get rid of. Can we somehow use `\OCA\Mail\IMAP\MessageMapper::getRawAttachments` or add a similar method instead? It should do the as `$folder->getAttachment` but with the new (testable) code architecture

Reply:
of course. I will do this

Reply:
should be good now